build/v1/types.go

```diff
@@ -1022,23 +1026,51 @@ type ImageChangeTrigger struct {
 	Paused bool `json:"paused,omitempty" protobuf:"varint,3,opt,name=paused"`
 }
 
+// ImageStreamTagReference captures the required elements for identifying the ImageStreamTag referenced by the more
+// generic ObjectReference BuildTriggerPolicy.ImageChange.From.  It is used by ImageChangeTriggerStatus, where a
+// specific instance of ImageChangeTriggerStatus in maintained in BuildConfigStatus.ImageChangeTriggers for each entry
+// in the BuildConfigSpec.Triggers array where the BuildTriggerPolicy.ImageChange pointer is set to a non-nil value
```

Comment:
Likewise, too many golang details for API documentation. The gory details can live here in the PR and any git commits.

```suggestion
// ImageStreamTagReference references the ImageStreamTag used in an image change trigger by namespace and name.
```